pkg/scalers/external_scaler.go

```diff
@@ -49,13 +50,19 @@ var externalLog = logf.Log.WithName("external_scaler")
 // NewExternalScaler creates a new external scaler - calls the GRPC interface
 // to create a new scaler
 func NewExternalScaler(config *ScalerConfig) (Scaler, error) {
+	metricType, err := GetExternalMetricTargetType(config.MetricType)
+	if err != nil {
+		return nil, fmt.Errorf("error getting external scaler metric type: %s", err)
+	}
```

Comment:
I don't get why we need this :/
We are using the same type that we receive, right? With default value and validations, but that could be done inline. Have you got plans to extend it to other scalers?

Reply:
Yes, I planned to extend it to all other scalers, was just waiting for some feedback before I go ahead and do that. See comment no. 5 in the PR description. Does that make sense?

Reply:
Hey @JorTurFer, @zroubalik. Going back to this after some absence period. Does this change in the external scaler make sense to you? If so, do you recommend doing the same for all other scalers?